code/services/LDAPService.php

```diff
@@ -880,18 +881,36 @@ public function setPassword(Member $member, $password)
         }
 
         try {
-            $this->update(
-                $userData['distinguishedname'],
-                array('unicodePwd' => iconv('UTF-8', 'UTF-16LE', sprintf('"%s"', $password)))
-            );
-
+            if ($oldPassword) {
+                $this->gateway->changePassword($userData['distinguishedname'], $password, $oldPassword);
+            } else {
+                $this->update(
```

Comment:
@halkyon Hey Sean, do you think it's worth adding a "resetPassword" method to LDAPGateway, now that we have "changePassword"?